aiida/orm/querybuilder.py

```diff
@@ -1994,15 +1995,20 @@ def get_query(self):
                 self._hash = queryhelp_hash
         return query
 
-    @staticmethod
-    def get_aiida_entity_res(value):
+    def get_aiida_entity_res(self, value):
         """Convert a projected query result to front end class if it is an instance of a `BackendEntity`.
 
         Values that are not an `BackendEntity` instance will be returned unaltered
 
         :param value: a projected query result to convert
         :return: the converted value
         """
+        if isinstance(value, collections.abc.Mapping):
+            return {key: self.get_aiida_entity_res(sub_value) for key, sub_value in value.items()}
```

Comment:
I wonder, should this not rather be fixed here:
https://github.com/aiidateam/aiida-core/blob/81d736c3c2f98c3f5ac02a3e6b21224e2e5bedf0/aiida/orm/convert.py#L26-L28

Also, it looks very similar, so I wonder why this is not working...
Is just the list/tuple stuff missing from there?

Reply:
You are right, it should be handled there but the problem is that there a `TypeError` can be raised if some of the elements are not node types, which is then caught by `get_aiida_entity_res` and it then returns the original value which is the original dictionary with unconverted values. I will fix it correctly